pkg/controllers/node/health/suite_test.go

```diff
@@ -312,9 +312,42 @@ var _ = Describe("Node Health", func() {
 
 			fakeClock.Step(60 * time.Minute)
 
-			// Determine to delete unhealthy nodes
+			// Determine if we should delete unhealthy nodes
 			for i := range 4 {
-				ExpectObjectReconciled(ctx, env.Client, healthController, nodes[i])
+				result := ExpectObjectReconciled(ctx, env.Client, healthController, nodes[i])
+				if i < 3 {
+					Expect(result.RequeueAfter).To(BeNumerically("~", time.Minute*5, time.Second))
+				}
+				nodeClaim = ExpectExists(ctx, env.Client, nodeClaims[i])
+				Expect(nodeClaim.DeletionTimestamp).To(BeNil())
+			}
+		})
+		It("should ignore unhealthy nodes if more then 20% of the nodes are unhealthy in a cluster", func() {
+			ExpectApplied(ctx, env.Client, nodePool)
+			nodeClaims := []*v1.NodeClaim{}
+			nodes := []*corev1.Node{}
+			for i := range 10 {
```

Comment:
Same question here -- I feel like the way that this is written is a bit odd -- is there a way to simplify some of this?